Sources/Helpers/AssociatedCellItemFactory.swift

```diff
@@ -0,0 +1,133 @@
+    
+    /// Returns an instance of `UniversalCollectionViewCellItem` and associates provided handlers with them
+    ///
+    /// - Parameters:
+    ///    - object: an object to create a cell item for it
+    ///    - index: the index of the object in the array
+    public func makeUniversalCellItem(object: U, index: Int) -> CollectionViewCellItem {
+        let cellItem = UniversalCollectionViewCellItem<T>()
+        cellItem.configurationHandler = { [weak self] cell in
+            guard let self = self else {
+                return
+            }
+            guard let configurationHandler = self.cellConfigurationHandler else {
+                fatalError("configurationHandler property for the CellItemFactory should be assigned before")
+            }
+            configurationHandler(object, cell, cellItem)
+        }
+        cellItem.sizeConfigurationHandler = { [weak self] (collectionView, sectionItem) -> CGSize in
+            guard let self = self else {
+                return .zero
+            }
+            guard let sizeConfigurationHandler = self.sizeConfigurationHandler else {
+                fatalError("sizeConfigurationHandler property for the CellItemFactory should be assigned before")
+            }
+            return sizeConfigurationHandler(object, collectionView, sectionItem)
+        }
+        cellItemConfigurationHandler?(index, object, cellItem)
+        return cellItem
+    }
+    
+    /// Returns an instance of `UniversalCollectionViewCellItem` and associates provided handlers with them
+    ///
+    /// - Parameters:
+    ///    - configure: a cell configuration handler
+    ///    - size: a cell size configuration handler
+    public func makeCellItem(configure configurationHandler: @escaping (T) -> Void,
+                             size sizeConfigurationHandler: @escaping (UICollectionView, CollectionViewSectionItem) -> CGSize) -> CollectionViewCellItem {
+        let cellItem = UniversalCollectionViewCellItem<T>()
+        cellItem.configurationHandler = configurationHandler
+        cellItem.sizeConfigurationHandler = sizeConfigurationHandler
+        return cellItem
+    }
+}
+
+extension AssociatedCellItemFactory: CellItemFactory {
+    public func makeCellItems(array: [Any]) -> [CollectionViewCellItem] {
+        if let array = array as? [U] {
+            return makeCellItems(array: array)
+        }
+        return []
+    }
+    
+    public func makeCellItems(object: Any, index: Int) -> [CollectionViewCellItem] {
+        if let object = object as? U {
+            if let initializationHandler = self.initializationHandler {
+                return initializationHandler(index, object).compactMap { cellItem in
+                    cellItem
+                }
+            }
+            else {
+                return [makeUniversalCellItem(object: object, index: index)]
+            }
+        }
+        return []
+    }
+    
+    public func join(factory: CellItemFactory) -> CellItemFactory {
```

Comment:
This function has incorrect name. It should be called like this `factory(byJoining factory: CellItemFactory)` if it returns another `CellItemFactory`